File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/visit/VisitHistoryCreator.java
package com.depromeet.threedollar.domain.domain.visit;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class VisitHistoryCreator {

    public static VisitHistory create(Long storeId, Long userId, VisitType type, LocalDate dateOfVisit) {
        return VisitHistory.builder()
            .storeId(storeId)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/domain/userservice/visit/VisitHistoryCreator.kt
package com.depromeet.threedollar.domain.rds.domain.userservice.visit

import java.time.LocalDate
import com.depromeet.threedollar.domain.rds.domain.TestFixture
import com.depromeet.threedollar.domain.rds.domain.userservice.store.Store

@TestFixture
object VisitHistoryCreator {

    @JvmStatic
    fun create(
        store: Store,
        userId: Long,
        type: VisitType,
        dateOfVisit: LocalDate,
    ): VisitHistory {
        return VisitHistory.builder()
            .store(store)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build()
    }

}
